pymodbus/transaction.py

```diff
@@ -663,6 +663,7 @@ def processIncomingPacket(self, data, callback):
                     # Could be an error response
                     if len(self.__buffer):
                         # Possible error ???
+                       self.resetFrame()
```

Comment:
resetting frame here would make the processing frame in the next line redundant.